tests/sync/file.cpp

```diff
@@ -0,0 +1,208 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "sync_test_utils.hpp"
+
+#include "shared_realm.hpp"
+#include <realm/util/file.hpp>
+
+using namespace realm;
+using namespace realm::util;
+using File = realm::util::File;
+
+static const std::string base_path = "/tmp/realm_objectstore_sync_file/";
```

Comment:
This should use `$TMPDIR` rather than being hardcoded to `/tmp` as `TestFile` does.